target/linux/x86/base-files/lib/preinit/43_grow_rootfspart

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+get_squashfs_size() {
+	local low high
+	low=$(hexdump -s 40 -n 4 -e '1/4 "%08x" "\n"' "$1")
+	high=$(hexdump -s 44 -n 4 -e '1/4 "%08x" "\n"' "$1")
+	echo $((0x$high$low))
+}
+
+run_grow_root_part() {
+    [ "$FAILSAFE" != "true" ] || return
+
+    local diskdev squashfs_magic
+    local SQUASHFS_ALIGN=$((0x10000))
+
+    . /lib/upgrade/common.sh
+
+    if export_bootdevice && export_partdevice diskdev 0; then
+	diskdev=/dev/$diskdev
+        squashfs_magic=$(dd bs=4 count=1 if="${diskdev}2" 2>/dev/null)
```

Comment:
logic wont handle mmc devs ( = /dev/mmcblk02 != /dev/mmcblk0p2 )

Reply:
I didn't think about mmc devices for x86. We also have nvm (nvm0n0p2). Fixed